### DISCUSSION TITLE
Can´t find tpp file? Please help a noob installing it

### DISCUSSION THREAD

**Discussion by ChotaFallen** [oddbear/GoXLR-TouchPortal-Plugin#4] [General]
Hey oddbear,

I really would like trying you plugin, but don´t know how. Even your readme confuses me.

Which is the correct file I have to download?
I downloaded "GoXLR-TouchPortal-Plugin-master" extracted it, but can´t find you goxlrworker.exe or entry.tp. in it there are ".cs" files.
You wrote download the tpp file and extract it, but there is no?

Could you please direct me to the correct files.

**Comment by oddbear** [comment]
Under the releases tab and 0.2, you will find: TouchPortal.GoXLR.Plugin.tpp
https://github.com/oddbear/GoXLR-TouchPortal-Plugin/releases

You also would need the  .NET Runtime 5.0.0 or higher installed for it to work.
https://dotnet.microsoft.com/download

Please tell if there is any way of making this easier to understand. All feedback is welcome 😄

**Comment by ChotaFallen** [comment]
Ahh ok thank you. Maybe I am too new to github. I expected the file via "download button" or in the list. 
![goxlr](https://user-images.githubusercontent.com/77210555/104108892-ccedb780-52c8-11eb-9d6e-df60f85c3990.png)

You could write in the readme, which file you should download like "v0.2-alpha"
But I found it.
I try it tomorrow.

oK installed it.
What can I do with it? I can´t see any changes in toucportal. Thought I see a list like the for OBS? How can I controll my goxlr with it now?

**Reply by oddbear** [reply]
The date of the sourcecode is a little newer than the latest build. Therefor one says 3 days ago and the build 30 Nov.

There are two things you an do, update routing TouchPortal will look something like this:
![image](https://user-images.githubusercontent.com/2109782/104110467-4b9e2100-52d8-11eb-9c74-5ff681a4b5c7.png)

And change profile:
![image](https://user-images.githubusercontent.com/2109782/104110478-6375a500-52d8-11eb-9518-ce57069ff5bc.png)

There are two ways of changing profiles. One where you manually enter the profile name in the action. And one that updates the list of profiles when the GoXLR App connects to the plugin. If the list of "GoXLR Profile Select" is empty, that means that the GoXLR app has not successfully connected to the plugin.

**Reply by ChotaFallen** [reply]
Thank you. I now have  your described problem that the GoXLR is not correct connected. Profile is empty (and I tried the routing table change, didn´t work either cause of same reason I think).
I reconnect the streamdeck app in GoXLR under "system/global settings/connect", Closed the GoXLR app and restart it, while touchportal is still running. 
Nothing changed hmmm
the dotnet is installed.

And I am right, I can´t see in the GoXLR app, when I changed maybe MIC in routing table from headphones on/off?

**Reply by oddbear** [reply]
This seems like you have done everything.

If you run the command `dotnet --list-runtimes` from `cmd.exe` in windows (terminal). Then you should see version 5.0.0 or higher in the list.
Other things that can be a issue:
- You have the official streamdeck plugin running (or any other application using the same port). The plugins including the official one, can only use one single port, so only one can be running at the same time. (can be tested with `Get-Process -Id (Get-NetTCPConnection -RemotePort 6805).OwningProcess` and `Get-Process -Id (Get-NetTCPConnection -LocalPort 6805).OwningProcess`).
- The TouchPortal plugin is on a different computer, should work. But not tested.

Under releases there is a version called `v0.1-alpha dev tooling` and a package `GUI.Dev.Client.Simulator.zip` this one can be used to fake a GoXLR, and should give more information.

I am also on discord, if you want a 1:1 screen share or similar. There is quite easy for me to miss a couple of things, or did not explain very good.

And maybe in the next version I should add a status in the Plugin for showing connected clients etc. 😄

**Reply by ChotaFallen** [reply]
Get-Process -Id (Get-NetTCPConnection -RemotePort 6805).OwningProcess` and `Get-Process -Id (Get-NetTCPConnection -LocalPort 6805).OwningProcess`).
The TouchPortal plugin is on a different computer, should work. But not tested.


> I am also on discord, if you want a 1:1 screen share or similar. There is quite easy for me to miss a couple of things, or did not explain very good.

The order 'Get' is not found :-(

Maybe we can try it via discord
Let me know when and where you are online.

**Comment by ChotaFallen** [comment]
Thank you very much for the great help via team viewer. Started my computer today, and it works like a charme. 👏👍😍

**Comment by masseyB** [comment]
Hey I need some help with your software and more info and am potentially wanting to hire you. Whats the best way to reach you?

**Reply by oddbear** [reply]
@masseyB I might be able to help, but not for hire.

If it's questions around the development or issues of this plugin, it might be best to start a new discussion thread on this github.

If it's about usage of TouchPortal or plugins in general, there is a discord server for TouchPortal with both support see [Discord here](https://www.touch-portal.com/support/). There is a lot of active people that will help there.

For usage of the plugin, we can take it here, or the goxlr channel on the TouchPortal discord.

For plugin development in general, the best place to start is the discord server.

_... For now I will log of for tonight, as I have long day tomorrow._

**Comment by masseyB** [comment]
I'm pretty new to this and I tried installing touch portal and got the
folders as you said. but I'm not sure I did the configuration right because
I don't see anything for GoXLR under the touch-portal apps.

I have a small software startup and we've been building some stream tech
and the original dev I had for it is unavailable. So either need to rebuild
or get someone else to tweak his code. Our goal is to get a partnership
with GoXLR and have our software installed with their products. But the
touch portal plugin may be a good first step. But as someone who has worked
with connecting software to GoXLR software, you seem to be an expert. Could
we schedule a call so I can maybe understand a bit more about how things
work? I can compensate you for your time.

Bests,
massey and gethr team
www.gethr.to

**Reply by oddbear** [reply]
I might help on some call today, but I am quite busy until around 18:00 CEST / GMT+2 / Norwegian time.
That is 16:00 GMT/UTC.
Or in about in 7 hours from now.

**Comment by masseyB** [comment]
Sorry, I was on holiday this weekend. Are you free today or tomorrow?

Bests,
massey and gethr team
www.gethr.to
